File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l=0;
        unordered_map<char, int> temp;
        int f=0;
        int maxN=0;
        for(int r=0; r<s.size(); r++){
            temp[s[r]]++;
            f = max(f, temp[s[r]]);
            if((r-l+1)-f>k){
                temp[s[l]]--;
                l++;
            }
            maxN = max(r-l+1, maxN);
        }
        return maxN;
    }
};

File: Data Structures & Algorithms/capacity-to-ship-packages-within-d-days/submission-2.py
class Solution:
    def shipWithinDays(self, weights: List[int], days: int) -> int:
        l = max(weights)
        r = sum(weights)
        res = r
        while l<=r:
            m = l + ((r-l)//2)
            count = 1
            temp = 0
            for weight in weights:
                if temp + weight > m:
                    count += 1
                    temp = weight
                else:
                    temp += weight
            if count > days:
                l = m+1
            else:
                r = m-1
                res = m
                
        return res